main/lsp/notifications/cancel_request.h

```diff
@@ -0,0 +1,21 @@
+#ifndef RUBY_TYPER_LSP_NOTIFICATIONS_CANCEL_REQUEST_H
+#define RUBY_TYPER_LSP_NOTIFICATIONS_CANCEL_REQUEST_H
+
+#include "main/lsp/LSPTask.h"
+
+namespace sorbet::realmain::lsp {
+class CancelRequestTask final : public LSPTask {
+    const std::unique_ptr<CancelParams> params;
+
+public:
+    CancelRequestTask(const LSPConfiguration &config, std::unique_ptr<CancelParams> params);
```

Comment:
The old special-cased cancelation path from LSPPreprocessor is now neatly encapsulated in this task.